nodes/modifier_change/offset_line.py

```diff
@@ -0,0 +1,210 @@
+# ##### BEGIN GPL LICENSE BLOCK #####
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either version 2
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software Foundation,
+#  Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# ##### END GPL LICENSE BLOCK #####
+
+import bpy
+from bpy.props import FloatProperty
+from sverchok.node_tree import SverchCustomTreeNode
+from sverchok.data_structure import updateNode
+
+
+class SvOffsetLineNode(bpy.types.Node, SverchCustomTreeNode):
+    """
+    Triggers: Offset Line 2D
+    Tooltip: Offsetting a Line into 2D space
+
+    Only X and Y dimensions of input points will be taken for work.
+    """
+    bl_idname = 'SvOffsetLineNode'
+    bl_label = 'Offset line'
+    bl_icon = 'OUTLINER_OB_EMPTY'
+
+    offset = FloatProperty(
+        name='offset', description='distance of offset',
+        default=0.1,
+        options={'ANIMATABLE'}, update=updateNode)
+
+    def sv_init(self, context):
+        self.inputs.new('VerticesSocket', 'Vers')
+        self.inputs.new('StringsSocket', "Edgs")
+        self.inputs.new('StringsSocket', "Offset").prop_name = 'offset'
+        self.outputs.new('VerticesSocket', 'Vers')
+        self.outputs.new('StringsSocket', "Faces")
+        self.outputs.new('StringsSocket', "OuterEdges")
+        self.outputs.new('StringsSocket', "VersMask")
+
+    def process(self):
+
+        if not ((self.outputs['Vers'].is_linked or self.outputs['Faces'].is_linked
+              or self.outputs['OuterEdges'].is_linked or self.outputs['VersMask'].is_linked) 
+              and (self.inputs['Vers'].is_linked or self.inputs['Edgs'].is_linked)):
+            return
+
+        verts_in = self.inputs['Vers'].sv_get()[0]
+        edges_in = self.inputs['Edgs'].sv_get()[0]
+        shift = self.inputs['Offset'].sv_get()[0][0]
+        
+        # avoid zero offset
+        if not shift:
+            return
+        
+        verts_out = []
+        faces_out = []
+
+        from mathutils import Vector, Matrix
+        from math import radians, pi, sin
+        from mathutils.geometry import normal
```

Comment:
stick imports at the top of the file, around the bpy import. See how other nodes do this.  

- 1.  standard python imports, (math, collections... etc)
- 2. bpy/blender imports, (mathutils...etc)
- 3. sverchok imports (utils..etc)